stanford_courses.module

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/**
+ * @file
+ * Stanford_courses.module.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Entity\Display\EntityViewDisplayInterface;
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Cache\Cache;
+
+/**
+ * Implements hook_page_attachments().
+ */
+function stanford_courses_page_attachments(array &$attachments) {
+  // Get the route.
+  $route = \Drupal::routeMatch()->getRouteName();
+  if (str_contains($route, 'page_view_courses_list_page_courses_list_page')) {
+    $attachments['#attached']['library'][] = 'stanford_courses/courses_page';
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_view().
+ */
+function stanford_courses_node_view(array &$build, EntityInterface $entity, EntityViewDisplayInterface $display, $view_mode) {
+  // Only run on stanford course items.
+  if ($entity->bundle() !== "stanford_course" || !node_is_page($entity)) {
+    return;
+  }
+
+  $rh_action = $entity->get('rh_action')->getString();
+  $rh_redirect = $entity->get('rh_redirect')->getString();
+
+  // Display a message to the user that this page would redirect to another
+  // location for other users.
+  if ($rh_action == "page_redirect" && !empty($rh_redirect)) {
+    \Drupal::messenger()
+      ->addWarning(t('This page will redirect to @url for other visitors.', ['@url' => $rh_redirect]));
+  }
```

Comment:
Do we not have this standardized somewhere?

Reply:
No, we do not, AFAIK, since I got this out of `stanford_events`, and it appears there and in `stanford_news`.  A prime candidate for refactor with the monorepo model, yeah?